### PR TITLE
WIP: Document KEP 3157 (watch-list) for milestone 1.30

### PR DESCRIPTION
**Pull request by p0lyn0mial**
TODO: Document `WatchListClient` as well.

### CODE REVIEW

File: content/en/docs/reference/command-line-tools-reference/feature-gates/watch-list.md [hunk 1]
```diff
@@ -9,5 +9,8 @@ stages:
   - stage: alpha 
     defaultValue: false
     fromVersion: "1.27"
+  - stage: beta
+    defaultValue: true
+    fromVersion: "1.30"
 ---
-Enable support for [streaming initial state of objects in watch requests](/docs/reference/using-api/api-concepts/#streaming-lists).
+Enable support for [streaming initial state of objects in watch requests](/content/en/docs/reference/using-api/api-concepts.md/#streaming-lists).
```
Comment (p0lyn0mial) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/watch-list.md:16]: the link was broken.

Comment (dipesh-rawat) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/watch-list.md:16]: The suggested link update seems unnecessary and incorrect. The existing link works as expected when rendered on the website. I'm not sure if this update is needed. 
Please check it out here: [https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/#feature-gates) _(check "WatchList" under the "List of feature gates" section)_.

The final redirection location is https://kubernetes.io/docs/reference/using-api/api-concepts/#streaming-lists.

Comment (sftim) [reply content/en/docs/reference/command-line-tools-reference/feature-gates/watch-list.md:16]:
```suggestion
Enable support for [streaming initial state of objects in watch requests](/docs/reference/using-api/api-concepts/#streaming-lists).
```

File: content/en/docs/reference/command-line-tools-reference/feature-gates/watch-list.md [hunk 2]
```diff
@@ -9,5 +9,8 @@ stages:
   - stage: alpha 
     defaultValue: false
     fromVersion: "1.27"
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates/watch-list.md:11]:
```suggestion
    fromVersion: "1.27"
    toVersion: "1.29"
```
The website does not work out `toVersion` for you, you need to be explicit.

**Comment by sftim** [comment]
/hold
pending assignment of publication date

/sig api-machinery

**Comment by Princesso** [comment]
Hello /sig api-machinery :wave: please provide a final review for this PR by March 26th 2024 18:00 PT to get this into the release. Thank you!
cc: @sftim

**Comment by p0lyn0mial** [comment]
The streaming feature didn't make it to 1.30. Hopefully, the feature will be completed in the next release, and I will be able to update this PR.

**Comment by sftim** [comment]
/remove-area blog

I'll close the pull request. You are very welcome to open a new PR against the development branch for a future release (and to reuse the work you did here).

/close

**Comment by drewhagen** [comment]
Heads up that `dev-1.31` branch is created now, if you choose to reopen this for the new release.

**Pull request closed without merging**